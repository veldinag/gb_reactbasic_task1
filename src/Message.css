.message {
    background-color: white;
    color: blue;
    font-size: 28px;
    font-weight: bold;
    padding: 10px;
}
.message span {
    color: green;
}