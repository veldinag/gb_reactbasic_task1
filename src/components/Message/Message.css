.fromMe {
    display: flex;
    width: 100%;
    justify-content: flex-end;
}

.toMe {
    display: flex;
    width: 100%;
    justify-content: flex-start;
}