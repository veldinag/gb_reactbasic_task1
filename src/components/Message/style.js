import {makeStyles} from "@material-ui/core";

const useStyles = makeStyles((theme) => ({
  msgBox: {
    padding: theme.spacing(1),
    marginBottom: theme.spacing(1),
    minWidth: "200px",
    backgroundColor: theme.palette.background.default
  }
}))

export default useStyles