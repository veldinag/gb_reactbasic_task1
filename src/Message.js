import './Message.css';

function Message(props) {
    return (
        <div>
            <h1 className="message">В модуль передан следующий текст: <span>{props.message}</span></h1>
        </div>
    );
}

export default Message;
