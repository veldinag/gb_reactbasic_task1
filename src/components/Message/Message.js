import {makeStyles, Paper, Typography} from "@material-ui/core"
import './Message.css'

const useStyles = makeStyles((theme) => ({
    msgBox: {
        padding: theme.spacing(1),
        marginBottom: theme.spacing(1),
        minWidth: "200px",
        backgroundColor: theme.palette.background.default
    }
}))

const Message = (props) => {
    const classes = useStyles()

    return (
        <div className={props.msg.type}>
            <Paper className={classes.msgBox}>
                <Typography display="block" variant="caption" color="secondary">{props.msg.date}</Typography>
                <Typography display="block" variant="body1" align="center">{props.msg.text}</Typography>
            </Paper>
        </div>
    )
}

export default Message;