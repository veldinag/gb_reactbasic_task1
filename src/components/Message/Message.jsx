import {Paper, Typography} from "@material-ui/core";
import './Message.css';
import useStyles from "./style";
import {useSelector} from "react-redux";
import {nameSelector} from "../../store/profile/selectors";

const Message = (props) => {
  const classes = useStyles();
  const name = useSelector(nameSelector);

  return (
    <div className={props.message.style}>
      <Paper className={classes.msgBox}>
        <Typography
          display="block"
          variant="caption"
          color="secondary">
          {"Date: " + props.message.date}
        </Typography>
        <Typography
          display="block"
          variant="caption"
          color="secondary">
          {"Author: " + name}
        </Typography>

        <Typography
          display="block"
          variant="body1"
          align="center">
          {props.message.message}
        </Typography>
      </Paper>
    </div>
  )
}

export default Message