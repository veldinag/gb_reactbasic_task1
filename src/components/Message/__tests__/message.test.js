import Message from "../Message";
import {render} from "@testing-library/react";

it('matches snapshot with props', () => {
   const message = {
      id: "001",
      date: "14.09.2021 12:22",
      message: "some message",
      style: "some style",
   }
   const component = render(
     <Message message={message} />
   );
   expect(component).toMatchSnapshot();
});