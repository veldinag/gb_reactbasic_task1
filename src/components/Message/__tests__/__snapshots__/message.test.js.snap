// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot with props 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="some style"
      >
        <div
          class="MuiPaper-root makeStyles-msgBox-1 MuiPaper-elevation1 MuiPaper-rounded"
        >
          <span
            class="MuiTypography-root MuiTypography-caption MuiTypography-colorSecondary MuiTypography-displayBlock"
          >
            14.09.2021 12:22
          </span>
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-alignCenter MuiTypography-displayBlock"
          >
            some message
          </p>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="some style"
    >
      <div
        class="MuiPaper-root makeStyles-msgBox-1 MuiPaper-elevation1 MuiPaper-rounded"
      >
        <span
          class="MuiTypography-root MuiTypography-caption MuiTypography-colorSecondary MuiTypography-displayBlock"
        >
          14.09.2021 12:22
        </span>
        <p
          class="MuiTypography-root MuiTypography-body1 MuiTypography-alignCenter MuiTypography-displayBlock"
        >
          some message
        </p>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
